Option to clear Sync server data and local browser state

ayumi opened the pull request:
Fix #7275

Fix https://github.com/brave/sync/issues/51

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:
1. Spawn a clean pyramid 0. Enable Sync and add bookmarks.
2. Spawn a clean pyramid 1 (`npm run start2`). Add it to 0's Sync profile and see the bookmarks sync over.
3. Save the Sync code words. We'll use this to demonstrate data was purged.
4. In Pyramid 0, go to Preferences > Sync and click Reset data. In the console you should see "Deleting user!!".
5. Close Pyramid 1, delete its user data, and spawn it clean again. Using the saved code words, try to re-add it to 0's deleted Sync profile. No bookmarks should sync over.
6. Back in Pyramid 0, you can use Sync freshly. Create a new profile and you should get a new set of code words.

Pix
![screen shot 2017-02-16 at 16 48 00](https://cloud.githubusercontent.com/assets/521861/23188721/a41ed65a-f844-11e6-9a25-3699d65b7d30.png)
![screen shot 2017-02-16 at 16 48 15](https://cloud.githubusercontent.com/assets/521861/23188724/a72f5cac-f844-11e6-9d83-a28dacd1edb0.png)

diracdeltas commented on app/renderer/components/preferences/syncTab.js:
```diff
@@ -242,6 +274,14 @@ class SyncTab extends ImmutableComponent {
       }
+      {
+        this.isSetup
+          ? <div class='syncClearData'>
```
i think react is throwing an error because it expects className here

diracdeltas commented on app/sync.js:
```diff
@@ -250,10 +254,20 @@ module.exports.onSyncReady = (isFirstRun, e) => {
+    if (dispatcherCallback) {
+      dispatcherCallback({actionType: syncConstants.SYNC_DELETE_USER})
+    }
+    appActions.changeSetting(settings.SYNC_ENABLED, false)
```
i would wait a short amount of time to call this to make sure that SYNC_DELETE_USER gets processed before the extension is disabled

ayumi replied:
could also add an IPC message for DELETED_USER and wait for that

diracdeltas replied:
that works too

diracdeltas commented:
test plan worked for me. :+1:

ayumi commented:
@diracdeltas have addressed comments.

deleting data requires sync to be enabled because it needs a valid AWS credential to issue the delete request. i've added a little message to explain:

<img width="342" alt="screen shot 2017-02-21 at 21 29 03" src="https://cloud.githubusercontent.com/assets/521861/23198716/5c5e23ae-f87e-11e6-8c69-e2a99bd6c7fe.png">
